app/_src/guides/progressively-rolling-out-unified-resource-naming.md

```diff
@@ -0,0 +1,164 @@
+```
+
+is not intuitive and does not point cleanly back to the right {{ Kuma }} resource. Different resources and their related stats often look unrelated, even when they describe the same traffic path.
+
+Starting with 2.12, you can adopt a unified resource naming scheme that makes names predictable, consistent, and directly tied to {{ Kuma }} resources. This improves observability, simplifies queries, and makes it much easier to understand what is happening in the mesh.
+
+## Prerequisites
+
+You have completed the [Kubernetes Quickstart]({{ docs }}/quickstart/kubernetes-demo/) and have the demo environment running. This guide assumes you already have the `demo-app` service running from the quickstart.
+
+<!-- vale Google.Headings = NO -->
+## Step 1: Create a ContainerPatch
+<!-- vale Google.Headings = YES -->
+
+Create and apply a `ContainerPatch` resource that enables unified naming on the sidecar:
+
+```yaml
+# containerpatch-unified-naming.yaml
+apiVersion: kuma.io/v1alpha1
+kind: ContainerPatch
+metadata:
+  name: enable-feature-unified-resource-naming
+  namespace: {{ kuma-system }}
+spec:
+  sidecarPatch:
+  - op: add
+    path: /env/-
+    value: '{
+      "name": "KUMA_DATAPLANE_RUNTIME_UNIFIED_RESOURCE_NAMING_ENABLED",
+      "value": "true"
+    }'
+```
+
+```sh
+kubectl apply -f containerpatch-unified-naming.yaml
+```
+
+This patch configures every sidecar that references it to set an environment variable that turns on the unified naming feature.
+
+<!-- vale Google.Headings = NO -->
+## Step 2: Enable for a single workload
+<!-- vale Google.Headings = YES -->
+
+Apply the patch to a workload by annotating its deployment. This lets you enable the feature progressively, service by service.
+
+```sh
+kubectl annotate -n kuma-demo deploy/demo-app kuma.io/container-patches=enable-feature-unified-resource-naming --overwrite
+```
+
+To disable later for that workload:
+
+```sh
+# set to an empty list
+kubectl annotate -n kuma-demo deploy/demo-app kuma.io/container-patches='' --overwrite
+
+# or remove the annotation entirely
+kubectl annotate -n kuma-demo deploy/demo-app kuma.io/container-patches-
+```
+
+<!-- vale Google.Headings = NO -->
+## Step 3: Verify the new naming
+<!-- vale Google.Headings = YES -->
+
+Inspect the sidecar stats to confirm that unified naming is applied:
+
+```sh
+kubectl exec -it -n kuma-demo deploy/demo-app -- curl -s localhost:9901/stats | grep -i kri
+```
+
+You should see entries that map directly to {{ Kuma }} resources, for example:
+
+```text
+cluster.kri_msvc_default_us-east-2_kuma-demo_demo-app_http
+```
+
+These names show the `MeshService` resource (`demo-app`) and section (`http`) clearly, making them much easier to connect back to the original {{ Kuma }} resource.
+
+You can also look at cluster names for confirmation:
+
+```sh
+kubectl exec -it -n kuma-demo deploy/demo-app -- curl -s localhost:9901/clusters | head -n 50
+```
+
+<!-- vale Google.Headings = NO -->
+## Step 4: Progressively roll out
```

Comment:
I don't think we need this section. You previously mentioned how to disable container patch